src/EditorFeatures/CSharpTest/IntroduceUsingStatement/IntroduceUsingStatementTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.IntroduceUsingStatement;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.IntroduceUsingStatement
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceUsingStatement)]
+    public sealed class IntroduceUsingStatementTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+            => new CSharpIntroduceUsingStatementCodeRefactoringProvider();
+
+        [Theory]
+        [InlineData("[|\r\n        var name = disposable; |]")]
+        [InlineData("v[||]ar name = disposable;")]
+        [InlineData("var[||] name = disposable;")]
+        [InlineData("var [||]name = disposable;")]
+        [InlineData("var na[||]me = disposable;")]
+        [InlineData("var name[||] = disposable;")]
+        [InlineData("var name [||]= disposable;")]
+        [InlineData("var name =[||] disposable;")]
+        [InlineData("var name = [||]disposable;")]
+        [InlineData("[|var name = disposable;|]")]
+        [InlineData("var name = disposable[||];")]
+        [InlineData("var name = disposable;[||]")]
+        [InlineData("var name = disposable[||]")]
+        public async Task RefactoringIsAvailable(string declaration)
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(System.IDisposable disposable)
+    {
+        " + declaration + @"
+    }
+}",
+@"class C
+{
+    void M(System.IDisposable disposable)
+    {
+        using (var name = disposable)
+        {
+        }
+    }
+}");
+        }
+
+        [Theory]
+        [InlineData("var name = d[||]isposable;")]
+        [InlineData("var name = disposabl[||]e;")]
+        [InlineData("var name=[|disposable|];")]
+        [InlineData("System.IDisposable name =[||]")]
+        public async Task RefactoringIsNotAvailable(string declaration)
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(System.IDisposable disposable)
+    {
+        " + declaration + @"
```

Comment:
is there a reason why this is explicitly filtered not to offer refactoring here?

Reply:
Yes. We were thinking that the refactoring is not relevant inside the initializer expression which may be arbitrarily complex,. The final example there has a missing initializer and I thought it would be okay to bail given invalid syntax, but this can always be expanded.